src/main/java/org/carlmontrobotics/robotcode2023/commands/RunRoller.java

```diff
@@ -43,12 +42,12 @@ public void end(boolean interrupted) {
     public boolean isFinished() {
         double time = timer.get();
 
-        if (!roller.hasGamePiece()) {
+        if (roller.hasGamePiece()) {
             timer.start();
         }
         SmartDashboard.putNumber("Time Target", mode.time);
         SmartDashboard.putNumber("SetRoller Time Elapsed (s)", time);
 
-        return !roller.hasGamePiece() && time > mode.time;
+        return roller.hasGamePiece() == mode.intake && time > mode.time;
     }
```

Comment:
Minor nit: What happens if `RunRoller` is started with `mode.intake == false` while there isn't a game piece in the rollers?

Reply:
I think that the command should run for the specified time and then terminate (I fixed the if condition). This should be fine. I'll also add a condition so that the command instantly terminates if the isFinished condition is met in `initialize()`.